### PR TITLE
Trigger diagnostic logs for the authentication script logs

### PR DESCRIPTION
**Pull request by chamathns**
### Proposed changes in this pull request
This is to publish diagnostic logs for the logs added in the authentication scripts by the user.

A sample log.info in authentication script :
```
var onLoginRequest = function(context) {
    Log.info('Executing step 1');
    .
    .
    .
}; 
```

the sample diagnostic log that's published:
```
{
    "logId": "f9983f88-6ae9-47a9-a486-9c4c44edd5c9",
    "recordedAt": {
        "seconds": 1662629738,
        "nanos": 596367000
    },
    "traceId": "9a297ce4-6c12-42dc-a5ed-d8beed653fd0",
    "resultStatus": "SUCCESS",
    "resultMessage": "Info log : Executing step 1",
    "actionId": "auth-script-logging",
    "componentId": "authentication-framework"
}
```
-

### When should this PR be merged
ASAP

### Follow up actions

- N/A

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [x] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [x] **Are all UI and API inputs run through forms or serializers?**
- [x] **Are all external inputs validated and sanitized appropriately?**
- [x] **Does all branching logic have a default case?**
- [x] **Does this solution handle outliers and edge cases gracefully?**
- [x] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java [hunk 1]
```diff
@@ -626,4 +626,15 @@ public static class AutoLoginConstant {
...
     }
+
+    /**
+     * Define logging constants for framework.
+     */
+    public static class LogConstants {
+
+        public static final String AUTHENTICATION_FRAMEWORK = "authentication-framework";
+        public static final String FAILED = "FAILED";
+        public static final String SUCCESS = "SUCCESS";
```
Comment (pulasthi7) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java:637]: Aren't these constants defined somewhere already?

Comment (chamathns) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java:637]: We have defined similar LogConstants in the oauth repo as well. 
So far, we have defined them component-wise. Not in a root-level repo

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/JsLogger.java [hunk 2]
```diff
@@ -31,54 +33,78 @@ public class JsLogger {
 
     private static JsLogger jsLogger = new JsLogger();
 
+    /**
+     * Returns an instance to log the javascript errors.
+     *
+     * @return
```
Comment (sahandilshan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/JsLogger.java:39]: Incompleted method comment

**Review by sahandilshan — CHANGES_REQUESTED** [review]

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java (hunk 1 above)
Comment (chamathns) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java]: Addressed with https://github.com/wso2/carbon-identity-framework/pull/4263/commits/ecab44c9e838c47a16f0b98beadde1dc03c7d291

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/JsLogger.java (hunk 2 above)
Comment (chamathns) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/JsLogger.java]: Addressed with https://github.com/wso2/carbon-identity-framework/pull/4263/commits/7ba7411de5a0e037de7a9c8656c58e7deab74708

**Review by sahandilshan — APPROVED** [review]

**Review by pulasthi7 — APPROVED** [review]

**Pull request merged**